lib/Auth/Process/AttributeAddUsersGroups.php

```diff
@@ -15,10 +15,39 @@
 use SimpleSAML\Assert\Assert;
 use SimpleSAML\Error;
 use SimpleSAML\Logger;
+use SimpleSAML\Module\ldap\Utils\Ldap as LdapUtils;
 use SimpleSAML\Utils;
+use Symfony\Component\Ldap\Adapter\ExtLdap\Query;
 
 class AttributeAddUsersGroups extends BaseFilter
 {
+    /** @var string */
+    protected string $searchUsername;
+
+    /** @var string */
+    protected string $searchPassword;
+
+    /** @var string */
+    protected string $product;
+
+
+    /**
+     * Initialize this filter.
+     *
+     * @param array $config Configuration information about this filter.
+     * @param mixed $reserved For future use.
+     */
+    public function __construct(array $config, $reserved)
+    {
+        parent::__construct($config, $reserved);
+
+        // Get filter specific config options
+        $this->searchUsername = $this->config->getString('search.username');
+        $this->searchPassword = $this->config->getString('search.password', null);
+        $this->product = $this->config->getString('ldap.product', 'ActiveDirectory');
```

Comment:
Why does this setting detault to `ActiveDirectory` and not to generic? Seems generic is a good default if nothing is specified?

Reply:
Agreed